src/IRCNetwork.cpp

```diff
@@ -450,8 +468,43 @@ bool CIRCNetwork::ParseConfig(CConfig* pConfig, CString& sError,
                 sModName = "modules_online";
             }
 
-            CString sModRet;
-            CString sArgs = sValue.Token(1, true);
+            // XXX Legacy crap
+            if (sModName == "sasl") {
+                if(ModuleInConfig(pConfig, "authtoservices")) {
+                    CUtils::PrintAction(
+                        "NOTICE: [sasl] is replaced by [authtoservices], "
+                        "so not loading [sasl]");
+                    continue;
+                } else {
+                    sNotice =
+                        "NOTICE: [sasl] was merged into [authtoservices], "
+                        "loading that with [sasl]'s configuration instead";
+                    sModName = "authtoservices";
+                    sArgs = "as_sasl";
+                }
+            }
+
+            if (sModName == "nickserv") {
+                if (ModuleInConfig(pConfig, "sasl")) {
+                    CUtils::PrintAction(
+                        "NOTICE: [nickserv] was merged into [authtoservices], "
```

Comment:
What if sasl module was loaded, but not configured - e.g. user never told sasl the password.